akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala

```diff
@@ -24,9 +24,14 @@ private[akka] final case class CompressionTable[T](version: Int, map: Map[T, Int
       require(map.values.sum + map.size == expectedGaplessSum, "Given compression map does not seem to be gap-less and starting from zero, " +
         "which makes compressing it into an Array difficult, bailing out! Map was: " + map)
 
-      val vals = map.toList.sortBy(_._2).iterator.map(_._1)
       val dtab = Array.ofDim[Object](map.size).asInstanceOf[Array[T]]
-      vals.copyToArray(dtab) // TODO HEAVY, AVOID COPYING AND THE MAP ETC!!!
+
+      val vals = map.toList.sortBy(_._2).iterator.zipWithIndex
```

Comment:
Ok, can do the array sort here. This is done very rarely btw, but good point anyway